src/compas/geometry/primitives/polyline.py

```diff
@@ -258,11 +259,123 @@ def transform(self, T):
             self.points[index].y = point[1]
             self.points[index].z = point[2]
 
+    def split_at_corners(self, angle_threshold):
+        """Splits a polyline at corners larger than the given angle_threshold
 
+        Parameters:
+        -----------
+        angle_threshold : float
+            In radians.
+
+        Returns
+        -------
+        list of :class:`compas.geometry.Polyline`
+
+        """
+
+        corner_ids = []
+        split_polylines = []
+        points = self.points
+        seg_ids = list(range(len(self.lines)))
+
+        if self.is_closed():
+            seg_ids.append(0)
+
+        for seg1, seg2 in pairwise(seg_ids):
+            angle = self.lines[seg1].vector.angle(self.lines[seg2].vector)
+            if angle >= angle_threshold:
+                corner_ids.append(seg1+1)
+
+        if self.is_closed() and len(corner_ids) > 0:
+            if corner_ids[-1] != len(points):
+                corner_ids = [corner_ids[-1]] + corner_ids
+        else:
+            corner_ids = [0] + corner_ids + [len(points)]
+
+        for id1, id2 in pairwise(corner_ids):
+            if id1 < id2:
+                split_polylines.append(Polyline(points[id1:id2+1]))
+            else:
+                looped_pts = [points[i] for i in range(id1, len(points))] + points[1:id2+1]
+                split_polylines.append(Polyline(looped_pts))
+
+        if self.is_closed() and not corner_ids:
+            return [Polyline(self.points)]
+
+        return split_polylines
+
+    def tangent_at_point_on_polyline(self, point):
+        """Calculates the tangent vector of a point on a polyline
+
+        Parameters:
+        -----------
+        point: :class:`compas.geometry.Point`
+
+        Returns
+        -------
+        :class:`compas.geometry.Vector`
+        """
+        for line in self.lines:
+            if is_point_on_line(point, line):
+                return line.direction
+            raise Exception('{} not found!'.format(point))
```

Comment:
I think this needs to have one less level indentation.  Sorry!
```suggestion
        raise Exception('{} not found!'.format(point))
```